postgis/lwgeom_transform.c

```diff
@@ -225,3 +226,215 @@ Datum LWGEOM_asKML(PG_FUNCTION_ARGS)
 		PG_RETURN_TEXT_P(kml);
 	PG_RETURN_NULL();
 }
+
+
+/********************************************************************************
+ * PROJ database reading functions
+ */
+
+
+#if POSTGIS_PROJ_VERSION >= 61
+
+struct srs_entry {
+	text* auth_name;
+	text* auth_code;
+};
+
+struct srs_data {
+	struct srs_entry* entries;
+	uint32_t num_entries;
+	uint32_t capacity;
+	uint32_t current_entry;
+};
+
+static Datum
+srs_tuple_from_entry(const struct srs_entry* entry, TupleDesc tuple_desc)
+{
+	HeapTuple tuple;
+	Datum tuple_data[4] = {0, 0, 0, 0};
+	bool  tuple_null[4] = {false, false, false, false};
+	PJ_CONTEXT *ctx = NULL;
+	const char* const empty_options[2] = {NULL};
+	const char* const wkt_options[2] = {"MULTILINE=NO", NULL};
+	const char *srtext;
+	const char *proj4text;
+
+	PJ *obj = proj_create_from_database(ctx,
+		text_to_cstring(entry->auth_name),
+		text_to_cstring(entry->auth_code),
+		PJ_CATEGORY_CRS, 0, empty_options);
+
+	if (!obj)
+		return (Datum) 0;
+
+	srtext = proj_as_wkt(ctx, obj, PJ_WKT1_GDAL, wkt_options);
+	proj4text = proj_as_proj_string(ctx, obj, PJ_PROJ_4, empty_options);
+
+	if (entry->auth_name)
+		tuple_data[0] = PointerGetDatum(entry->auth_name);
+	else
+		tuple_null[0] = true;
+
+	if (entry->auth_code)
+		tuple_data[1] = PointerGetDatum(entry->auth_code);
+	else
+		tuple_null[1] = true;
+
+	if (srtext)
+		tuple_data[2] = PointerGetDatum(cstring_to_text(srtext));
+	else
+		tuple_null[2] = true;
+
+	if (proj4text)
+		tuple_data[3] = PointerGetDatum(cstring_to_text(proj4text));
+	else
+		tuple_null[3] = true;
+
+	tuple = heap_form_tuple(tuple_desc, tuple_data, tuple_null);
+	proj_destroy(obj);
+
+	return HeapTupleGetDatum(tuple);
+}
+
+#endif
+
+/**
+ * Search for srtext and proj4text given auth_name and auth_srid,
+ * returns TABLE(auth_name text, auth_srid text, srtext text, proj4text text)
+ */
+Datum postgis_srs_entry(PG_FUNCTION_ARGS);
+PG_FUNCTION_INFO_V1(postgis_srs_entry);
+Datum postgis_srs_entry(PG_FUNCTION_ARGS)
+{
+#if POSTGIS_PROJ_VERSION < 61
+	elog(ERROR, "%s is not supported with Proj < 6.1", __func__);
+#else
+	Datum result;
+	struct srs_entry entry;
+	text* auth_name = PG_GETARG_TEXT_P(0);
+	text* auth_code = PG_GETARG_TEXT_P(1);
+	TupleDesc tuple_desc;
+
+	if (get_call_result_type(fcinfo, 0, &tuple_desc) != TYPEFUNC_COMPOSITE) {
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+			errmsg("%s called with incompatible return type", __func__)));
+	}
+	BlessTupleDesc(tuple_desc);
+
+	entry.auth_name = auth_name;
+	entry.auth_code = auth_code;
+	result = srs_tuple_from_entry(&entry, tuple_desc);
+
+	if (result)
+		PG_RETURN_DATUM(srs_tuple_from_entry(&entry, tuple_desc));
+	else
+		PG_RETURN_NULL();
+#endif
+}
+
+
+Datum postgis_srs_entry_all(PG_FUNCTION_ARGS);
+PG_FUNCTION_INFO_V1(postgis_srs_entry_all);
+Datum postgis_srs_entry_all(PG_FUNCTION_ARGS)
+{
+#if POSTGIS_PROJ_VERSION < 61
```

Comment:
Shouldn't this be 

#if POSTGIS_PROJ_VERSION < 60

per @rouault

Reply:
yes all of that should work with 6.0